openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ReturnTypeGenerator.java

```diff
@@ -287,7 +291,11 @@ private TypeDescriptorNode handleMultipleResponse(ApiResponses responses)
                 TypeDescriptorNode record;
                 record = returnNode.orElseGet(
                         () -> createSimpleNameReferenceNode(createIdentifierToken(ANYDATA)));
-                if (responseCode.equals(GeneratorConstants.HTTP_200)) {
+                //Check the default behaviour for return type according to POST method.
+                boolean isWithOutStatusCode =
+                        (httpMethod.equals(POST) && responseCode.equals(GeneratorConstants.HTTP_201)) ||
+                        (!httpMethod.equals(POST) && responseCode.equals(GeneratorConstants.HTTP_200));
```

Comment:
Shouldn't we have the L297 condition as below or, am I missing something?
```suggestion
                        (httpMethod.equals(POST) && responseCode.equals(GeneratorConstants.HTTP_201)) ||
                        (httpMethod.equals(OK) && responseCode.equals(GeneratorConstants.HTTP_200));
```

Reply:
This condition is for checking the two conditions that the return type can only be with its schema type without its status code.
1. when the POST method has status code 201 (Accepted)
2. when the other methods have status code 200(Ok) (line 297)
Therefore Line297 we need to check whether it is not POST and with 200. the suggested condition we are unable to use it because it checks the Ok that is already a 200 status code but needs to replace by checking the http method. 🙂